src/Chapter3.hs

```diff
@@ -479,6 +505,43 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
+data MagicalCity = MagicalCity {
+  magicalCityCastle :: Castle,
+  magicalCityFacility :: Facility,
+  magicalCityHouse :: [House]
+} deriving (Show)
+
+data Castle = NoneCastle 
+            | Castle { castleName :: String } 
+            | CastleWithWall { castleName :: String } 
+            deriving (Show)
+
+data Facility = NoneFacility | Church | Library deriving (Show)
+
+data House = House {
+  housePeople :: Int
+} deriving (Show)
```

Comment:
Nice idea to have a number of people as `Int` field!

However, this allows having any number of people in any House (even negative), though we have restrictions on types of houses 🏘️ 

This could be fixed by introducing another type to control the number of people 🙂

Reply:
I added a new type meaning restrictions of the number of people in one house.